invokeai/app/datatypes/image.py

```diff
@@ -0,0 +1,38 @@
+from enum import Enum
+from typing import Optional
+from pydantic import BaseModel, Field
+
+from invokeai.app.datatypes.metadata import ImageMetadata
+
+
+class ImageType(str, Enum):
+    RESULT = "results"
+    INTERMEDIATE = "intermediates"
+    UPLOAD = "uploads"
+
+
+class ImageField(BaseModel):
+    """An image field used for passing image objects between invocations"""
+
+    image_type: str = Field(
+        default=ImageType.RESULT, description="The type of the image"
+    )
+    image_name: Optional[str] = Field(default=None, description="The name of the image")
+
+    class Config:
+        schema_extra = {
+            "required": [
+                "image_type",
+                "image_name",
+            ]
+        }
+
+
+class ImageResponse(BaseModel):
```

Comment:
This should go in the `/api` folder, since it's an api-only data model.

Reply:
Ok, so we can have a `datatypes` module in each folder. How about a single `datatypes` file in each folder, or would you prefer a datatypes folder with categorized files?

Reply:
made `api/datatypes/images.py`

Reply:
Hrm... I'm not sure if the convention here is the same, but I'd usually call these `models` or `viewmodels`, depending on usage. Might want `api/models/images.py`

Reply:
done!